### PR TITLE
Prepare the release of decompress.1.4.3 which remove the bigarray-compat dependency

### PR DESCRIPTION
**Pull request by dinosaure**
See this error:
```
- Error: This expression has type
-          Bigstringaf.t =
-            (char, Bigarray_compat.int8_unsigned_elt,
-             Bigarray_compat.c_layout)
-            Bigarray_compat.Array1.t
-        but an expression was expected of type
-          Zl.bigstring =
-            (char, Bigarray.int8_unsigned_elt, Bigarray.c_layout)
-            Bigarray.Array1.t
- (cd _build/default/fuzz && ./binary_search.exe)
- binary search: PASS
- 
[ERROR] The compilation of carton.0.4.3 failed at "dune runtest -p carton -j 31".
```

`decompress.1.4.3` deletes the `bigarray-compat` dependency and the result is an incompatibility with packages such as `carton`. /cc @kit-ty-kate

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
Could you past the opam output (lines prefixed by `#` and starting by `#=== ERROR while compiling <pkg> ==================================#`). It looks like an error during the tests, not during compilation and it would make it easier to check this fact if the log were there.

**Comment by dinosaure** [comment]
```
#=== ERROR while compiling carton.0.4.3 =======================================#
# context              2.1.2 | linux/x86_64 | ocaml-base-compiler.4.14.0 | file:///home/opam/opam-repository
# path                 ~/.opam/4.14/.opam-switch/build/carton.0.4.3
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune runtest -p carton -j 31
# exit-code            1
# env-file             ~/.opam/log/carton-2432-86956e.env
# output-file          ~/.opam/log/carton-2432-86956e.out
### output ###
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlc.opt -w -40 -g -bin-annot -I fuzz/.pack_headers.eobjs/byte -I /home/opam/.opam/4.14/lib/bigstringaf -I /home/opam/.opam/4.14/lib/crowbar -I /home/opam/.opam/4.14/lib/decompress/de -I /home/opam/.opam/4.14/lib/decompress/zl -I /home/opam/.opam/4.14/lib/fmt -I src/carton/.carton.objs/byte -no-alias-deps -o fuzz/.pack_headers.eobjs/byte/dune__exe__Pack_headers.cmo -c -impl fuzz/pack_headers.ml)
# File "fuzz/pack_headers.ml", line 28, characters 29-30:
# 28 |     Carton.Dec.make payload ~z ~allocate ~uid_ln:20
#                                   ^
# Error: This expression has type
#          Bigstringaf.t =
#            (char, Bigarray_compat.int8_unsigned_elt,
#             Bigarray_compat.c_layout)
#            Bigarray_compat.Array1.t
#        but an expression was expected of type
#          Zl.bigstring =
#            (char, Bigarray.int8_unsigned_elt, Bigarray.c_layout)
#            Bigarray.Array1.t
# (cd _build/default/fuzz && ./binary_search.exe)
# binary search: PASS 
```

File: packages/carton/carton.0.3.0/opam [hunk 1]
```diff
@@ -15,7 +15,7 @@ depends: [
   "dune" {>= "2.8.0"}
   "ke" {>= "0.4"}
   "duff" {>= "0.3"}
-  "decompress" {>= "1.3.0"}
+  "decompress" {>= "1.3.0" & < "1.4.3"}
```
Comment (kit-ty-kate) [inline packages/carton/carton.0.3.0/opam:18]:
```suggestion
  "decompress" {>= "1.3.0"}
  "decompress" {with-test & < "1.4.3"}
```

File: packages/carton/carton.0.4.0/opam [hunk 2]
```diff
@@ -15,7 +15,7 @@ depends: [
   "dune" {>= "2.8.0"}
   "ke" {>= "0.4"}
   "duff" {>= "0.3"}
-  "decompress" {>= "1.3.0"}
+  "decompress" {>= "1.3.0" & < "1.4.3"}
```
Comment (kit-ty-kate) [inline packages/carton/carton.0.4.0/opam:18]:
```suggestion
  "decompress" {>= "1.3.0"}
  "decompress" {with-test & < "1.4.3"}
```

File: packages/carton/carton.0.4.1/opam [hunk 3]
```diff
@@ -15,7 +15,7 @@ depends: [
   "dune" {>= "2.8.0"}
   "ke" {>= "0.4"}
   "duff" {>= "0.3"}
-  "decompress" {>= "1.3.0"}
+  "decompress" {>= "1.3.0" & < "1.4.3"}
```
Comment (kit-ty-kate) [inline packages/carton/carton.0.4.1/opam:18]:
```suggestion
  "decompress" {>= "1.3.0"}
  "decompress" {with-test & < "1.4.3"}
```

File: packages/carton/carton.0.4.2/opam [hunk 4]
```diff
@@ -15,7 +15,7 @@ depends: [
   "dune" {>= "2.8.0"}
   "ke" {>= "0.4"}
   "duff" {>= "0.3"}
-  "decompress" {>= "1.3.0"}
+  "decompress" {>= "1.3.0" & < "1.4.3"}
```
Comment (kit-ty-kate) [inline packages/carton/carton.0.4.2/opam:18]:
```suggestion
  "decompress" {>= "1.3.0"}
  "decompress" {with-test & < "1.4.3"}
```

File: packages/carton/carton.0.4.3/opam [hunk 5]
```diff
@@ -15,7 +15,7 @@ depends: [
   "dune" {>= "2.8.0"}
   "ke" {>= "0.4"}
   "duff" {>= "0.3"}
-  "decompress" {>= "1.4.1"}
+  "decompress" {>= "1.4.1" & < "1.4.3"}
```
Comment (kit-ty-kate) [inline packages/carton/carton.0.4.3/opam:18]:
```suggestion
  "decompress" {>= "1.4.1"}
  "decompress" {with-test & < "1.4.3"}
```

**Comment by dinosaure** [comment]
I added a constraint on `carton` about `cstruct.6.1.0` due to this error:
```
#=== ERROR while compiling carton.0.3.0 =======================================#
# context              2.1.2 | linux/x86_64 | ocaml-options-only-afl.1 | pinned(https://github.com/mirage/ocaml-git/releases/download/carton-v0.3.0/carton-carton-v0.3.0.tbz)
# path                 ~/.opam/4.14/.opam-switch/build/carton.0.3.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p carton -j 31
# exit-code            1
# env-file             ~/.opam/log/carton-14607-41945f.env
# output-file          ~/.opam/log/carton-14607-41945f.out
### output ###
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlc.opt -w -40 -g -bin-annot -I bin/carton/.hxd_cmdliner.objs/byte -I /home/opam/.opam/4.14/lib/cmdliner -I /home/opam/.opam/4.14/lib/fmt -I /home/opam/.opam/4.14/lib/hxd/core -I /home/opam/.opam/4.14/lib/result -I /home/opam/.opam/4.14/lib/rresult -no-alias-deps -o bin/carton/.hxd_cmdliner.objs/byte/hxd_cmdliner.cmo -c -impl bin/carton/hxd_cmdliner.ml)
# File "bin/carton/hxd_cmdliner.ml", line 124, characters 12-23:
# 124 |   let env = Arg.env_var "HXD_COLOR" in
#                   ^^^^^^^^^^^
# Alert deprecated: Cmdliner.Arg.env_var
# Use Cmd.Env.info instead.
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlopt.opt -w -40 -g -I bin/carton/.hxd_cmdliner.objs/byte -I bin/carton/.hxd_cmdliner.objs/native -I /home/opam/.opam/4.14/lib/cmdliner -I /home/opam/.opam/4.14/lib/fmt -I /home/opam/.opam/4.14/lib/hxd/core -I /home/opam/.opam/4.14/lib/result -I /home/opam/.opam/4.14/lib/rresult -intf-suffix .ml -no-alias-deps -o bin/carton/.hxd_cmdliner.objs/native/hxd_cmdliner.cmx -c -impl bin/carton/hxd_cmdliner.ml)
# File "bin/carton/hxd_cmdliner.ml", line 124, characters 12-23:
# 124 |   let env = Arg.env_var "HXD_COLOR" in
#                   ^^^^^^^^^^^
# Alert deprecated: Cmdliner.Arg.env_var
# Use Cmd.Env.info instead.
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlc.opt -w -40 -g -bin-annot -I src/carton/.thin.objs/byte -I /home/opam/.opam/4.14/lib/bigarray-compat -I /home/opam/.opam/4.14/lib/bigstringaf -I /home/opam/.opam/4.14/lib/checkseum -I /home/opam/.opam/4.14/lib/cstruct -I /home/opam/.opam/4.14/lib/decompress/de -I /home/opam/.opam/4.14/lib/decompress/zl -I /home/opam/.opam/4.14/lib/ke -I /home/opam/.opam/4.14/lib/logs -I /home/opam/.opam/4.14/lib/optint -I src/carton/.carton.objs/byte -intf-suffix .ml -no-alias-deps -o src/carton/.thin.objs/byte/thin.cmo -c -impl src/carton/thin.ml)
# File "src/carton/thin.ml", line 47, characters 25-36:
# 47 |           let len = min (Cstruct.len inputs) (Cstruct.len src) in
#                               ^^^^^^^^^^^
# Error: Unbound value Cstruct.len
# Hint: Did you mean lenv?
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlc.opt -w -40 -g -bin-annot -I bin/carton/.get.eobjs/byte -I /home/opam/.opam/4.14/lib/bigarray-compat -I /home/opam/.opam/4.14/lib/bigstringaf -I /home/opam/.opam/4.14/lib/bos -I /home/opam/.opam/4.14/lib/cmdliner -I /home/opam/.opam/4.14/lib/decompress/de -I /home/opam/.opam/4.14/lib/decompress/zl -I /home/opam/.opam/4.14/lib/digestif/c -I /home/opam/.opam/4.14/lib/fmt -I /home/opam/.opam/4.14/lib/fpath -I /home/opam/.opam/4.14/lib/hxd/core -I /home/opam/.opam/4.14/lib/hxd/string -I /home/opam/.opam/4.14/lib/mmap -I /home/opam/.opam/4.14/lib/result -I /home/opam/.opam/4.14/lib/rresult -I bin/carton/.hxd_cmdliner.objs/byte -I bin/fiber/.fiber.objs/byte -I src/carton/.carton.objs/byte -no-alias-deps -o bin/carton/.get.eobjs/byte/dune__exe__Get.cmo -c -impl bin/carton/get.ml)
# File "bin/carton/get.ml", line 169, characters 14-32:
# 169 |   let exits = Term.default_exits in
#                     ^^^^^^^^^^^^^^^^^^
# Alert deprecated: Cmdliner.Term.default_exits
# Use Cmd.Exit.defaults or Cmd.info's defaults ~exits value instead.
# File "bin/carton/get.ml", line 187, characters 4-13:
# 187 |     Term.info "get" ~doc ~exits ~man )
#           ^^^^^^^^^
# Alert deprecated: Cmdliner.Term.info
# Use Cmd.info instead.
# File "bin/carton/get.ml", line 189, characters 15-26:
# 189 | let () = Term.(exit_status @@ eval cmd)
#                      ^^^^^^^^^^^
# Alert deprecated: Cmdliner.Term.exit_status
# Use Stdlib.exit and Cmd.eval' instead.
# File "bin/carton/get.ml", line 189, characters 30-34:
# 189 | let () = Term.(exit_status @@ eval cmd)
#                                     ^^^^
# Alert deprecated: Cmdliner.Term.eval
# Use Cmd.v and one of Cmd.eval* instead.
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlc.opt -w -40 -thread -g -bin-annot -I bin/carton/.index_pack.eobjs/byte -I /home/opam/.opam/4.14/lib/bigarray-compat -I /home/opam/.opam/4.14/lib/bigstringaf -I /home/opam/.opam/4.14/lib/bos -I /home/opam/.opam/4.14/lib/cmdliner -I /home/opam/.opam/4.14/lib/cstruct -I /home/opam/.opam/4.14/lib/decompress/de -I /home/opam/.opam/4.14/lib/decompress/zl -I /home/opam/.opam/4.14/lib/digestif/c -I /home/opam/.opam/4.14/lib/fmt -I /home/opam/.opam/4.14/lib/fpath -I /home/opam/.opam/4.14/lib/ke -I /home/opam/.opam/4.14/lib/mmap -I /home/opam/.opam/4.14/lib/result -I /home/opam/.opam/4.14/lib/rresult -I bin/fiber/.fiber.objs/byte -I src/carton/.carton.objs/byte -I src/carton/.thin.objs/byte -no-alias-deps -o bin/carton/.index_pack.eobjs/byte/dune__exe__Index_pack.cmo -c -impl bin/carton/index_pack.ml)
# File "bin/carton/index_pack.ml", line 99, characters 23-34:
# 99 |         let len = min (Cstruct.len inputs) (Cstruct.len src) in
#                             ^^^^^^^^^^^
# Error: Unbound value Cstruct.len
# Hint: Did you mean lenv?
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlc.opt -w -40 -g -bin-annot -I bin/carton/.verify_pack.eobjs/byte -I /home/opam/.opam/4.14/lib/bigarray-compat -I /home/opam/.opam/4.14/lib/bigstringaf -I /home/opam/.opam/4.14/lib/bos -I /home/opam/.opam/4.14/lib/cmdliner -I /home/opam/.opam/4.14/lib/decompress/de -I /home/opam/.opam/4.14/lib/decompress/zl -I /home/opam/.opam/4.14/lib/digestif/c -I /home/opam/.opam/4.14/lib/fmt -I /home/opam/.opam/4.14/lib/fpath -I /home/opam/.opam/4.14/lib/mmap -I /home/opam/.opam/4.14/lib/result -I /home/opam/.opam/4.14/lib/rresult -I bin/fiber/.fiber.objs/byte -I src/carton/.carton.objs/byte -no-alias-deps -o bin/carton/.verify_pack.eobjs/byte/dune__exe__Verify_pack.cmo -c -impl bin/carton/verify_pack.ml)
# File "bin/carton/verify_pack.ml", line 301, characters 14-32:
# 301 |   let exits = Term.default_exits in
#                     ^^^^^^^^^^^^^^^^^^
# Alert deprecated: Cmdliner.Term.default_exits
# Use Cmd.Exit.defaults or Cmd.info's defaults ~exits value instead.
# File "bin/carton/verify_pack.ml", line 312, characters 4-13:
# 312 |     Term.info "verify-pack" ~doc ~exits ~man )
#           ^^^^^^^^^
# Alert deprecated: Cmdliner.Term.info
# Use Cmd.info instead.
# File "bin/carton/verify_pack.ml", line 314, characters 15-26:
# 314 | let () = Term.(exit_status @@ eval cmd)
#                      ^^^^^^^^^^^
# Alert deprecated: Cmdliner.Term.exit_status
# Use Stdlib.exit and Cmd.eval' instead.
# File "bin/carton/verify_pack.ml", line 314, characters 30-34:
# 314 | let () = Term.(exit_status @@ eval cmd)
#                                     ^^^^
# Alert deprecated: Cmdliner.Term.eval
# Use Cmd.v and one of Cmd.eval* instead.
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlopt.opt -w -40 -g -I src/carton/.thin.objs/byte -I src/carton/.thin.objs/native -I /home/opam/.opam/4.14/lib/bigarray-compat -I /home/opam/.opam/4.14/lib/bigstringaf -I /home/opam/.opam/4.14/lib/checkseum -I /home/opam/.opam/4.14/lib/cstruct -I /home/opam/.opam/4.14/lib/decompress/de -I /home/opam/.opam/4.14/lib/decompress/zl -I /home/opam/.opam/4.14/lib/ke -I /home/opam/.opam/4.14/lib/logs -I /home/opam/.opam/4.14/lib/optint -I src/carton/.carton.objs/byte -I src/carton/.carton.objs/native -intf-suffix .ml -no-alias-deps -o src/carton/.thin.objs/native/thin.cmx -c -impl src/carton/thin.ml)
# File "src/carton/thin.ml", line 47, characters 25-36:
# 47 |           let len = min (Cstruct.len inputs) (Cstruct.len src) in
#                               ^^^^^^^^^^^
# Error: Unbound value Cstruct.len
# Hint: Did you mean lenv?
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlopt.opt -w -40 -g -I bin/carton/.get.eobjs/byte -I bin/carton/.get.eobjs/native -I /home/opam/.opam/4.14/lib/bigarray-compat -I /home/opam/.opam/4.14/lib/bigstringaf -I /home/opam/.opam/4.14/lib/bos -I /home/opam/.opam/4.14/lib/cmdliner -I /home/opam/.opam/4.14/lib/decompress/de -I /home/opam/.opam/4.14/lib/decompress/zl -I /home/opam/.opam/4.14/lib/digestif/c -I /home/opam/.opam/4.14/lib/fmt -I /home/opam/.opam/4.14/lib/fpath -I /home/opam/.opam/4.14/lib/hxd/core -I /home/opam/.opam/4.14/lib/hxd/string -I /home/opam/.opam/4.14/lib/mmap -I /home/opam/.opam/4.14/lib/result -I /home/opam/.opam/4.14/lib/rresult -I bin/carton/.hxd_cmdliner.objs/byte -I bin/carton/.hxd_cmdliner.objs/native -I bin/fiber/.fiber.objs/byte -I bin/fiber/.fiber.objs/native -I src/carton/.carton.objs/byte -I src/carton/.carton.objs/native -intf-suffix .ml -no-alias-deps -o bin/carton/.get.eobjs/native/dune__exe__Get.cmx -c -impl bin/carton/get.ml)
# File "bin/carton/get.ml", line 169, characters 14-32:
# 169 |   let exits = Term.default_exits in
#                     ^^^^^^^^^^^^^^^^^^
# Alert deprecated: Cmdliner.Term.default_exits
# Use Cmd.Exit.defaults or Cmd.info's defaults ~exits value instead.
# File "bin/carton/get.ml", line 187, characters 4-13:
# 187 |     Term.info "get" ~doc ~exits ~man )
#           ^^^^^^^^^
# Alert deprecated: Cmdliner.Term.info
# Use Cmd.info instead.
# File "bin/carton/get.ml", line 189, characters 15-26:
# 189 | let () = Term.(exit_status @@ eval cmd)
#                      ^^^^^^^^^^^
# Alert deprecated: Cmdliner.Term.exit_status
# Use Stdlib.exit and Cmd.eval' instead.
# File "bin/carton/get.ml", line 189, characters 30-34:
# 189 | let () = Term.(exit_status @@ eval cmd)
#                                     ^^^^
# Alert deprecated: Cmdliner.Term.eval
# Use Cmd.v and one of Cmd.eval* instead.
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlopt.opt -w -40 -g -I bin/carton/.verify_pack.eobjs/byte -I bin/carton/.verify_pack.eobjs/native -I /home/opam/.opam/4.14/lib/bigarray-compat -I /home/opam/.opam/4.14/lib/bigstringaf -I /home/opam/.opam/4.14/lib/bos -I /home/opam/.opam/4.14/lib/cmdliner -I /home/opam/.opam/4.14/lib/decompress/de -I /home/opam/.opam/4.14/lib/decompress/zl -I /home/opam/.opam/4.14/lib/digestif/c -I /home/opam/.opam/4.14/lib/fmt -I /home/opam/.opam/4.14/lib/fpath -I /home/opam/.opam/4.14/lib/mmap -I /home/opam/.opam/4.14/lib/result -I /home/opam/.opam/4.14/lib/rresult -I bin/fiber/.fiber.objs/byte -I bin/fiber/.fiber.objs/native -I src/carton/.carton.objs/byte -I src/carton/.carton.objs/native -intf-suffix .ml -no-alias-deps -o bin/carton/.verify_pack.eobjs/native/dune__exe__Verify_pack.cmx -c -impl bin/carton/verify_pack.ml)
# File "bin/carton/verify_pack.ml", line 301, characters 14-32:
# 301 |   let exits = Term.default_exits in
#                     ^^^^^^^^^^^^^^^^^^
# Alert deprecated: Cmdliner.Term.default_exits
# Use Cmd.Exit.defaults or Cmd.info's defaults ~exits value instead.
# File "bin/carton/verify_pack.ml", line 312, characters 4-13:
# 312 |     Term.info "verify-pack" ~doc ~exits ~man )
#           ^^^^^^^^^
# Alert deprecated: Cmdliner.Term.info
# Use Cmd.info instead.
# File "bin/carton/verify_pack.ml", line 314, characters 15-26:
# 314 | let () = Term.(exit_status @@ eval cmd)
#                      ^^^^^^^^^^^
# Alert deprecated: Cmdliner.Term.exit_status
# Use Stdlib.exit and Cmd.eval' instead.
# File "bin/carton/verify_pack.ml", line 314, characters 30-34:
# 314 | let () = Term.(exit_status @@ eval cmd)
#                                     ^^^^
# Alert deprecated: Cmdliner.Term.eval
# Use Cmd.v and one of Cmd.eval* instead.
```

**Comment by dinosaure** [comment]
I added a constraint `"bigstringaf" {with-test & < "0.9.0"}` due to this error:
```
#=== ERROR while compiling carton.0.4.2 =======================================#
# context              2.1.2 | linux/x86_64 | ocaml-base-compiler.4.09.1 | pinned(https://github.com/mirage/ocaml-git/releases/download/carton-v0.4.2/carton-carton-v0.4.2.tbz)
# path                 ~/.opam/4.09/.opam-switch/build/carton.0.4.2
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune runtest -p carton -j 31
# exit-code            1
# env-file             ~/.opam/log/carton-2434-07c484.env
# output-file          ~/.opam/log/carton-2434-07c484.out
### output ###
# (cd _build/default && /home/opam/.opam/4.09/bin/ocamlc.opt -w -40 -g -bin-annot -I fuzz/.pack_headers.eobjs/byte -I /home/opam/.opam/4.09/lib/bigstringaf -I /home/opam/.opam/4.09/lib/crowbar -I /home/opam/.opam/4.09/lib/decompress/de -I /home/opam/.opam/4.09/lib/decompress/zl -I /home/opam/.opam/4.09/lib/fmt -I src/carton/.carton.objs/byte -no-alias-deps -o fuzz/.pack_headers.eobjs/byte/dune__exe__Pack_headers.cmo -c -impl fuzz/pack_headers.ml)
# File "fuzz/pack_headers.ml", line 28, characters 29-30:
# 28 |     Carton.Dec.make payload ~z ~allocate ~uid_ln:20
#                                   ^
# Error: This expression has type
#          Bigstringaf.t =
#            (char, Bigarray.int8_unsigned_elt, Bigarray.c_layout)
#            Bigarray.Array1.t
#        but an expression was expected of type
#          Zl.bigstring =
#            (char, Zl.Bigarray.int8_unsigned_elt, Zl.Bigarray.c_layout)
#            Bigarray_compat.Array1.t
# (cd _build/default/fuzz && ./binary_search.exe)
# binary search: PASS
```

**Pull request merged**